Pod/Classes/WPMediaGroupPickerViewController.m

```diff
@@ -52,7 +51,52 @@ - (void)viewDidLoad
 
 - (void)loadData
 {
+    [self.dataSource loadGroupDataWithSuccess:^{
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self.tableView reloadData];
+        });
+    } failure:^(NSError *error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [self showError:error];
+        });
+    }];
+}
+
+- (void)showError:(NSError *)error {
+    [self.refreshControl endRefreshing];
     [self.tableView reloadData];
+    NSString *title = NSLocalizedString(@"Media Library", @"Title for alert when a generic error happened when loading media");
+    NSString *message = NSLocalizedString(@"There was a problem when trying to access your media. Please try again later.",  @"Explaining to the user there was an generic error accesing media.");
+    NSString *cancelText = NSLocalizedString(@"OK", "");
+    NSString *otherButtonTitle = nil;
+    if (error.domain == WPMediaPickerErrorDomain &&
+        error.code == WPMediaErrorCodePermissionsFailed) {
+        otherButtonTitle = NSLocalizedString(@"Open Settings", @"Go to the settings app");
+        title = NSLocalizedString(@"Media Library", @"Title for alert when access to the media library is not granted by the user");
+        message = NSLocalizedString(@"This app needs permission to access your device media library in order to add photos and/or video to your posts. Please change the privacy settings if you wish to allow this.",
```

Comment:
I know we already use this text in the library, but we might want to check with @iamthomasbishop if it needs any adjustments – @iamthomasbishop, is the title and message text here okay for the 'you don't have permission' alert?